packages/react/src/Pagination/Pagination.tsx

```diff
@@ -56,13 +56,27 @@ export const Pagination = memo(
     hrefBuilder = defaultHrefBuilder,
     pageAttributesBuilder,
     marginPageCount = 1,
-    showPages,
+    showPages = true,
     surroundingPageCount = 2,
     'aria-label': ariaLabel,
     'data-testid': testId,
     ...rest
   }: PaginationProps) => {
-    const {isMedium} = useWindowSize()
+    if (typeof showPages === 'object') {
+      showPages = Object.values(showPages).some(x => x)
+
+      if (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test') {
+        // eslint-disable-next-line no-console
+        console.warn('The responsive object value for `showPages` is deprecated. Please use a boolean value instead.')
+      }
+    }
```

Comment:
I can see why this is useful, but do we need this? Guessing we did this in another component, and in hindsight think we should rely on putting a more comprehensive migration guide into the change log entry. That avoids us having to clean up logs like this down the line.